### ISSUE TITLE
Make error message when compiling without selecting a board more user friendly

### ISSUE THREAD

**Issue by per1234** [arduino/arduino-pro-ide#431]
**Is your feature request related to a problem? Please describe.**
When you attempt to compile without first selecting a board, a somewhat cryptic error message is displayed:
>Error: Request compile failed with message: 2 UNKNOWN: no FQBN provided

**To Reproduce**
1. Start Arduino Pro IDE. Don't select a board from the board menu.
2. **Sketch > Verify/Compile** - the message in question will be shown in a notification as well as in the "Output" pane.

**Describe the solution you'd like**
Show a user friendly error message. Something like:
>No board selected. Please select your Arduino board from the **Tools > Board** menu.

**Additional context**
I think this will be a fairly common occurrence since no board is selected even when you open a new sketch (https://github.com/arduino/arduino-pro-ide/issues/369, https://github.com/arduino/arduino-pro-ide/issues/338). Users transitioning from Arduino IDE to Arduino Pro IDE will be accustomed to their board selection being persistent.

Even among Arduino CLI users, I have found that the concept of "FQBN" is not widely known. IDE users are even less likely to be familiar with this acronym, and even less anxious to spend their time learning what it means.

**Comment by ubidefeo** [comment]
@per1234 
I agree on the more friendly messages, but I believe #369 and #338 should nonetheless be addressed
